draft-pda-protocol.md

```diff
@@ -529,81 +527,79 @@ example, that clients are prohibited from talking to helpers but not the leader.
 Is it OK that leaders learn that about a client? I'm not sure, so I'd be
 inclined to remove this unless we have a concrete use case.]
 
-Next, the client issues a POST request to `[leader]/upload_finish`. It begins by
-setting up an HPKE {{!I-D.irtf-cfrg-hpke}} context for the helper by running
+Next, the client issues a POST request to `[leader]/upload_finish`. The payload
+is structured as follows:
 
 ~~~
-enc, context = SetupBaseS(pk, [TODO])
+struct {
+  PATask task; // Equal to PAUploadStartReq.task
+  uint64 time; // UNIX time (in seconds).
+  PAEncryptedInputShare leader_input_share;
+  PAEncryptedInputShare helper_input_share;
+} PAUploadFinishReq;
```

Comment:
This seems like a good opportunity to just be:

```
PAEncryptedInputShare shares<...>;
```

That is like the least we can do to lay the groundwork for >1 helper.

Reply:
Done.